### ISSUE TITLE
Spam control

### ISSUE THREAD

**Issue by auryn-macmillan** [auryn-macmillan/DAOForum#5]
**How should spam be controlled?**

**Comment by ghost** [comment]
If a moderation team is opted for, this would probably be their raison d'être. I don't know enough about automated spam control to have an opinion one way or another, but I would guess it's tough to nail and will inevitably let some slip through the cracks while occasionally punishing actual users. Mods are probably the best solution for this, no need to reinvent the wheel.

**Comment by auryn-macmillan** [comment]
A simple automated spam control mechanism would be to have an insignificant cost to make a post, or to up/downvote. If you had to pay a cent to make a post, it would become cost prohibitive to mount large scale spam attacks.

**Comment by vmichalik** [comment]
yep I like the cost mechanism. it works off the same primitives used in cryptocurrency so it's neatly self-referential in that way

**Comment by ghost** [comment]
The cost mechanism is a great idea, but should we put an upper bound on it per user? For example if someone is incredibly active in the DAO, posting 50 times a day, even at 1 cent a post that will cost them ~$180 a year just to participate. The counter argument to that would be that if the upper bound is low enough, an attack might just pay the limit and spam away. 

Now that I think about it a cost mechanism would also disincentive low quality posts that aren't quite spam (i.e 'shitposting', the scourge of the web), and double as a revenue source for the platform/slockit DAO. Other DAOs would only pay a price that is perfectly proportional to their use of the platform.

**Comment by corpetty** [comment]
What if the cost mechanism took into account percentage of holdings by the poster?  This would incentivize people who are really invested, and want to say a lot, to invest into the platform, and keep people with minuscule amounts of tokens from spamming.

**Comment by FelixA** [comment]
So the more 'money' tokens you have, the less 'taxes' (posting fees) you have to pay? Sounds a bit like our tax system... 
What if we make a standard fee that is equal for everyone --> these fees are going into a pool --> then dynamically these fees are donated to posters that got the most likes - dislikes

**Comment by corpetty** [comment]
I was thinking more along the lines of limiting the rate of posting, instead of the fee associated with a single post.

**Comment by FelixA** [comment]
Don't get me wrong, it is a good idea to prevent spamming. but it also disincentivizes people from submitting good ideas without having a lot of tokens. my idea on the other hand would incentivize to create fake accounts and like your own posts very strongly. Both ideas have their pros and cons

**Comment by corpetty** [comment]
It doesn't quite disincentive people with small amounts of tokens to submit good ideas, it keeps them from posting rapidly, incentivizing thoughtful posts.  On the other hand, people with more tokens, presumably more involved, have the ability to post more often.  

This allows those with more "clout" to answer questions without real regard to losing their ability to contribute new ideas as well.  People under this type of limiting aren't limited to the amount of posts they are able to put out, only the time limit in which they are able to do so.  If a spammer is found, it can be dealt with quickly without too much getting through, while keeping the conversation at a good amount of intellectual depth.

**Comment by auryn-macmillan** [comment]
What if you had a relatively insignificant fee, 1 cent for example, to post a thread; this fee would be paid to the platform. Then a smaller fee to reply to/comment on a thread, say 0.1 cent, which would be paid to whoever made the thread, and a smaller still fee to up/downvote which would also be paid to the OP. Nested comments would work similar to a pyramid scheme, commentors would receive a small portion of the fees paid to all of their nested commentors.

This would create a net loss for spam posts/comments, but a net gain for good quality post.

**Comment by ghost** [comment]
@auryn-macmillan I think that's absolutely brilliant. Love the 'pyramid scheme' mechanism. Good contributors get rewarded and are incentivized to keep contributing, and spammers/bad posts will be rarer as they run the risk of getting punished with downvotes, but downvotes actually mean something in this set up. We could even take it a step further and have variable transaction fee pricing based on the thread type. For example, the official threads for contract proposals would logically be the most expensive to contribute to as we want those threads to have more substance and less filler. 

Systems that make extensive use of microtransactions haven't really caught on as much as some initially theorized, as the first step is generally 'first get some bitcoin'. We should take advantage of the fact that every end-user is by extension a crypto user, and are already open to these concepts to a certain degree. There's bound to be some more novel possibilities that weren't feasible with the standard forum structure. 

The next question would be - what are they paying with? Does daohub needs its own token? I can't imagine having the payments in ETH or BTC, as there's a psychological barrier with spending those that might just drive users away. Like how casinos use chips instead of cash.

**Comment by auryn-macmillan** [comment]
@cjharty cheers, I was pretty excited by the idea. I'm sure it's an amalgamation of other ideas I've heard around and by no means original. But I honestly couldn't point you in the direction of where I would have heard it.

I really like the idea of variable fees depending on the type of thread. Would you imagine this as something users could set when they create a thread?  Would rates be an integer or a binary option; $0.x cents to post or fees on/off? Bearing in mind that too much complexity will serve as a barrier to entry. I think part of what makes reddit so successful is it's simplicity.

As far as which token to use, that's a good question. I would be tempted to say just use ETH, but I see what you're saying about there being a mental barrier to spending "real" money. The issue with using a DAOHub specific token would be that it has to have a real value for the antispam mechanic to work. Perhaps you peg it to another currency, or hope that it naturally gains value through its utility and being exchanged freely. But ultimately, having to convert another crypto to the DAOHub token adds another layer of complexity / barrier to entry. I tend to think the key is to make the fee to post/comment/vote etc so low that it circumvents the majority of people's mental barrier to spending "real" money.
